clang/lib/CodeGen/CodeGenAction.cpp

```diff
@@ -806,6 +806,28 @@ void BackendConsumer::AspectMismatchDiagHandler(
   }
 }
 
+void BackendConsumer::SYCLIllegalVirtualCallDiagHandler(
+    const llvm::DiagnosticInfoIllegalVirtualCall &D) {
+  const llvm::SmallVector<std::pair<StringRef, unsigned>, 8> &CallChain =
+      D.getCallChain();
+  auto &KI = CallChain.front();
+
+  SourceLocation LocCookie = SourceLocation::getFromRawEncoding(KI.second);
+  assert(LocCookie.isValid() &&
+         "Invalid location for caller in aspect mismatch diagnostic");
+  Diags.Report(LocCookie, diag::err_sycl_illegal_virtual_call)
+      << llvm::demangle(KI.first.str());
+
+  for (size_t I = 1; I < CallChain.size(); ++I) {
+    auto &CalleeInfo = CallChain[I];
+    LocCookie = SourceLocation::getFromRawEncoding(CalleeInfo.second);
+    assert(LocCookie.isValid() &&
```

Comment:
Same here

Reply:
Fixed in d383c2c57147a8c9127f4ee91c8e72c8be0a4839